tests/ci/finish_check.py

```diff
@@ -20,27 +21,38 @@
 from synchronizer_utils import SYNC_BRANCH_PREFIX
 
 
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="Script to merge the given PR. Additional checks for approved "
+        "status and green commit statuses could be done",
+    )
+    parser.add_argument(
+        "--wf-status",
+        type=str,
+        default="",
+        help="overall workflow status [success|failure]",
+    )
+    return parser.parse_args()
+
+
 def main():
     logging.basicConfig(level=logging.INFO)
+    args = parse_args()
 
-    has_failure = False
-
-    # FIXME: temporary hack to fail Mergeable Check in MQ if pipeline has any failed jobs
-    if len(sys.argv) > 1 and sys.argv[1] == "--pipeline-failure":
-        has_failure = True
+    has_workflow_failures = args.wf_status == FAILURE
 
     pr_info = PRInfo(need_orgs=True)
     gh = Github(get_best_robot_token(), per_page=100)
     commit = get_commit(gh, pr_info.sha)
-    statuses = None
 
     if pr_info.is_merge_queue:
-        # in MQ Mergeable check status must never be green if any failures in workflow
-        if has_failure:
-            set_mergeable_check(commit, "workflow failed", "failure")
+        # in MQ Mergeable check status must never be green if any failures in the workflow
+        if has_workflow_failures:
+            set_mergeable_check(commit, "workflow failed", FAILURE)
         else:
             # This must be the only place where green MCheck is set in the MQ (in the end of CI) to avoid early merge
-            set_mergeable_check(commit, "workflow passed", "success")
+            set_mergeable_check(commit, "workflow passed", SUCCESS)
```

Comment:
```suggestion
            set_mergeable_check(commit, "workflow passed", SUCCESS)
            return
```

The script should finish here on the merge_queue. The following `else` can go away to improve readability.